src/wh_crypto.c

```diff
@@ -257,7 +257,36 @@ int wh_Crypto_MlDsaSerializeKeyDer(MlDsaKey* key, uint16_t max_size,
         return WH_ERROR_BADARGS;
     }
 
-    ret = wc_Dilithium_KeyToDer(key, buffer, max_size);
+    /* Choose appropriate serialization based on key flags */
+    if (key->prvKeySet && key->pubKeySet) {
+#if defined(WOLFSSL_DILITHIUM_PRIVATE_KEY) && \
+    defined(WOLFSSL_DILITHIUM_PUBLIC_KEY)
+        /* Full keypair - use KeyToDer */
+        ret = wc_Dilithium_KeyToDer(key, buffer, max_size);
+#else
+        ret = WH_ERROR_NOHANDLER;
```

Comment:
Bad args instead?  This is a locally detected error.  No handler is expected as a server response when the client doesn't know any better.

Reply:
Yeah you are right. I feel like we need a "not compiled in" error, to somehow signify that "user input is valid (e.g. not malformed) but not supported by the current build configuration". I'll change to BADARGS for now.